projects/vdk-plugins/vdk-control-api-auth/src/vdk/plugin/control_api_auth/base_auth.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import json
+import logging
+import time
+from typing import Optional
+from typing import Union
+
+from requests import post
+from requests.auth import HTTPBasicAuth
+from requests_oauthlib import OAuth2Session
+from vdk.plugin.control_api_auth.auth_config import AuthConfig
+from vdk.plugin.control_api_auth.auth_config import AuthConfigFolder
+from vdk.plugin.control_api_auth.auth_config import InMemAuthConfiguration
+from vdk.plugin.control_api_auth.auth_exception import VDKAuthException
+from vdk.plugin.control_api_auth.auth_request_values import AuthRequestValues
+from vdk.plugin.control_api_auth.login_types import LoginTypes
+
+log = logging.getLogger(__name__)
+
+
+class AuthenticationCache:
+    """
+    Class used to keep the fields in the vdk-cred.json file from which
+    authorization and authentication tokens can be taken. Used only when
+    caching is enabled.
+    """
+
+    def __init__(
+        self,
+        authorization_url=None,
+        refresh_token=None,
+        access_token=None,
+        access_token_expiration_time=0,
+        client_id=None,
```

Comment:
why None in favour of priorly used string placeholder by `auth.py`?

Reply:
To get rid of the codacy warning messages.